programs/solana/programs/ics26-router/src/state.rs

```diff
@@ -32,12 +32,9 @@ pub struct CounterpartyInfo {
     /// Client ID on the counterparty chain
     #[max_len(MAX_CLIENT_ID_LENGTH)]
     pub client_id: String,
-    /// Connection ID on the counterparty chain
-    #[max_len(MAX_CLIENT_ID_LENGTH)]
-    pub connection_id: String,
     /// Merkle prefix for proof verification
-    #[max_len(128)]
-    pub merkle_prefix: Vec<u8>,
+    #[max_len(16, 128)]
```

Comment:
I'm not sure if this is too large. In practice, the largest we have ever used is 2. So I could reduce this to 4 or 8 to be on the safe side